skipruntime-ts/adapters/postgres/src/index.ts

```diff
@@ -171,35 +168,26 @@ FOR EACH ROW EXECUTE FUNCTION %I();`,
       }
     };
 
-    const setup = async () => {
-      await initData().catch(
-        error(
-          `Uncaught error during Skip async initialization for Postgres table ${table}:`,
-        ),
-      );
-
-      this.client.on("notification", (msg) => {
-        if (msg.channel == instance && msg.payload !== undefined) {
-          const query = key.select(table, msg.payload);
-          this.client.query(query).then(
-            (changes) => {
-              const k = key.type == "TEXT" ? msg.payload! : Number(msg.payload);
-              callbacks.update([[k, changes.rows as Json[]]], false);
-            },
-            error(`Error executing Postgres query "${query}":`),
-          );
-        }
-      });
-      await setupPgNotify().catch(
-        error(`Uncaught error setting up Postgres triggers on ${table}:`),
-      );
-    };
-
-    setup().catch(
-      error(
-        `Uncaught error during async Skip update of Postgres table ${table}`,
-      ),
-    );
+    await initData();
+
+    this.client.on("notification", (msg) => {
+      if (
+        msg.channel == instance &&
+        msg.payload !== undefined &&
+        this.open_instances.has(instance)
+      ) {
+        const query = key.select(table, msg.payload);
+        const run = async () => {
+          const changes = await this.client.query(query);
+          const k = key.type == "TEXT" ? msg.payload! : Number(msg.payload);
+          await callbacks.update([[k, changes.rows as Json[]]], false);
+        };
+        run().catch((e: unknown) => {
+          error(`Error executing Postgres query "${query}":`)(e);
+        });
```

Comment:
I think this will incorrectly report errors executing queries when the query succeeds but `callbacks.update` throws because of some mapper downstream.

Maybe something like the following?

```suggestion
        const k = key.type == "TEXT" ? msg.payload! : Number(msg.payload);
        const run = async () => {
          const changes = await this.client.query(query).catch(error(`Error executing Postgres query "${query}":`))
          await callbacks.update([[k, changes.rows as Json[]]], false);
        };
        run().catch(error(`Uncaught error triggered by Postgres update (key ${k}, table ${table}):`));
```